apps/src/maze/maze.js

```diff
@@ -444,29 +329,41 @@ Maze.init = function (config) {
       Blockly.JavaScript.INFINITE_LOOP_TRAP = codegen.loopHighlight("Maze");
     }
 
-    Maze.start_ = undefined;
-    Maze.finish_ = undefined;
+    Maze.subtype.start = undefined;
+    Maze.subtype.finish = undefined;
 
     // Locate the start and finish squares.
     for (var y = 0; y < Maze.map.ROWS; y++) {
       for (var x = 0; x < Maze.map.COLS; x++) {
         var cell = Maze.map.getTile(y, x);
         if (cell === SquareType.START) {
-          Maze.start_ = {x: x, y: y};
+          Maze.subtype.start = {x: x, y: y};
         } else if (cell === SquareType.FINISH) {
-          Maze.finish_ = {x: x, y: y};
+          Maze.subtype.finish = {x: x, y: y};
         } else if (cell === SquareType.STARTANDFINISH) {
-          Maze.start_ = {x: x, y: y};
-          Maze.finish_ = {x: x, y: y};
+          Maze.subtype.start = {x: x, y: y};
+          Maze.subtype.finish = {x: x, y: y};
```

Comment:
There's definitely more refactoring to do but I'm tempted to save it for a future PR!